bots/heroes/hunter_main.lua

```diff
@@ -0,0 +1,374 @@
+				bActionTaken = core.OrderAbilityEntity(botBrain, abilSilence, unitTarget)
+			end
+		end
+	end
+	
+	--Hemorrhage
+	if not bActionTaken and abilHemorrhage:CanActivate() and nIsSighted then -- Cast Hermorrhage if
+		local nRange = abilHemorrhage:GetRange()
+		if nTargetDistanceSq < (nRange * nRange) then
+			if (nLastHarassUtility > botBrain.nHemorrhageThreshold) then -- target is in close range and ability threshold
+				bActionTaken = core.OrderAbilityEntity(botBrain, abilHemorrhage, unitTarget)
+			elseif (nEffectiveHP < 350) then -- or ability would kill a hero
+				bActionTaken = core.OrderAbilityEntity(botBrain, abilHemorrhage, unitTarget)
+			end
+		end
+	end
+
+	--codex
+	if not bActionTaken and itemCodex and core.nDifficulty ~= core.nEASY_DIFFICULTY then --Difficulty modifier taken from Parasite Bot made by Mellow_Ink and Kairus101
+		if nIsSighted and itemCodex:CanActivate() then -- Cast Codex if
+			local nRange = itemCodex:GetRange()
+			local nCodexDamage = ((itemCodex:GetLevel() * 100) + 300)
+			if nTargetDistanceSq < (nRange * nRange) then --Codex selected and Target is in range and 
+				if (core.NumberElements(tLocalEnemyHeroes) > 2) and (core.NumberElements(tLocalAllyHeroes) > 2) then -- Cast in a team fight (more than 2 enemy and 2 ally heroes)
+					bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemCodex, unitTarget)
+				elseif (nEffectiveHP < nCodexDamage * 1.75) and (nEffectiveHP > nCodexDamage) then -- Casting will wound the target, but not kill it
+					bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemCodex, unitTarget)
+				end
+			end
+		end
+	end
+	
+	-- Silence A -Helpful-
+	if  not bActionTaken and itemNullBlade then --If BH has Nullfire blade
+		local nRange = itemNullBlade:GetRange()
+		local nInRange = nTargetDistanceSq < (nRange * nRange)
+		if abilSilence:CanActivate() and (SilenceLvl > 2) and itemNullBlade:CanActivate() then --  Silence can activate and has Nullfire to remove debuff and will provide a decent buff
+			if (unitSelf:GetHealthPercent() > .25) then -- Or BH is in fighting order with more damage then 
+				bActionTaken = core.OrderAbilityEntity(botBrain, abilSilence, unitSelf) -- Cast Silence on self 
+			end
+			
+		elseif itemNullBlade:CanActivate() then
+			if unitSelf:HasState(State_Hunter_Ability1_Buff) then -- Verifies there is a debuff on BH
+				bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemNullBlade, unitSelf) -- Purges the silence debuff
+			end
+		elseif nInRange and (nLastHarassUtility > botBrain.nManaBurn1Threshold) then --Enemy is in range and nullfire passes threshold check
+				bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemNullBlade, unitTarget) --Casts Nullfire blade on enemy.
+		end
+	end
+	
+	--feast
+	if not bActionTaken and abilFeast:CanActivate() then
+		local nRange = abilFeast:GetRange()
+		local vecEnemyCreepPosition = unitECreep:GetPosition()
```

Comment:
Gives error as it is list of units not a unit. You have to check positions of all of them.